airbyte-workers/src/main/java/io/airbyte/workers/internal/EmptyAirbyteSource.java

```diff
@@ -50,43 +55,35 @@ public EmptyAirbyteSource() {
   @Override
   public void start(final WorkerSourceConfig workerSourceConfig, final Path jobRoot) throws Exception {
 
-    if (workerSourceConfig == null || workerSourceConfig.getSourceConnectionConfiguration() == null) {
-      // TODO: When the jobConfig is fully updated and tested, we can remove this extra check that makes
-      // us compatible with running a reset with
-      // a null config
-      isResetBasedForConfig = false;
-    } else {
-      final ResetSourceConfiguration resetSourceConfiguration;
-      try {
-        resetSourceConfiguration =
-            Jsons.object(workerSourceConfig.getSourceConnectionConfiguration(), ResetSourceConfiguration.class);
-        streamsToReset.addAll(resetSourceConfiguration.getStreamsToReset());
-      } catch (final IllegalArgumentException e) {
-        log.error("The configuration provided to the reset has an invalid format");
-        throw new IllegalArgumentException("The reset configuration format is invalid", e);
-      }
+    // TODO: When the jobConfig is fully updated and tested, we can remove this extra check that makes
+    // us compatible with running a reset with a null config
+    isResetBasedForConfig = workerSourceConfig != null && workerSourceConfig.getSourceConnectionConfiguration() != null && !streamsToReset.isEmpty();
```

Comment:
I don't think that this condition will work since streamsToReset will always be empty. because it is initialize on line 64.

Reply:
good point.